File: backend/credit-service/src/main/java/com/bank/credit/dto/CreateTariffRequest.java
package com.bank.credit.dto;

import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import java.math.BigDecimal;

public record CreateTariffRequest(
        @NotBlank(message = "Название тарифа обязательно")
        String name,

        @NotNull(message = "Процентная ставка обязательна")
        @DecimalMin(value = "0.01", message = "Ставка должна быть положительной")
        BigDecimal interestRate,

        BigDecimal minAmount,
        BigDecimal maxAmount,

        @Min(value = 1, message = "Минимальный срок — 1 день")
        Integer minTermDays,

        Integer maxTermDays
) {}

File: ios/BankShared/Sources/BankShared/DTOs/Requests/CreateTariffRequest.swift
import Foundation

public struct CreateTariffRequest: Codable, Sendable {
    public let name: String
    public let interestRate: Decimal
    public let minAmount: Decimal?
    public let maxAmount: Decimal?
    public let minTermDays: Int?
    public let maxTermDays: Int?

    public init(name: String, interestRate: Decimal, minAmount: Decimal? = nil, maxAmount: Decimal? = nil, minTermDays: Int? = nil, maxTermDays: Int? = nil) {
        self.name = name
        self.interestRate = interestRate
        self.minAmount = minAmount
        self.maxAmount = maxAmount
        self.minTermDays = minTermDays
        self.maxTermDays = maxTermDays
    }
}
